packages/app/src/vis-packs/core/scatter/MappedScatterVis.tsx

```diff
@@ -1,34 +1,42 @@
 import { ScatterVis, useDomain, useSafeDomain, useVisDomain } from '@h5web/lib';
-import { toArray } from '@h5web/lib/src/vis/utils';
+import type { ScatterAxisParams } from '@h5web/lib/src/vis/scatter/models';
 import type { NumArray, ScaleType } from '@h5web/shared';
-import { assertDefined } from '@h5web/shared';
 import { createPortal } from 'react-dom';
 import shallow from 'zustand/shallow';
 
 import { useBaseArray } from '../hooks';
-import type { AxisMapping } from '../models';
 import { DEFAULT_DOMAIN } from '../utils';
 import ScatterToolbar from './ScatterToolbar';
 import { useScatterConfig } from './config';
 
 interface Props {
   value: NumArray;
-  axisMapping: AxisMapping;
+  abscissaParams: ScatterAxisParams;
+  ordinateParams: ScatterAxisParams;
   title: string;
   toolbarContainer: HTMLDivElement | undefined;
   colorScaleType?: ScaleType;
 }
 
 function MappedScatterVis(props: Props) {
-  const { value, axisMapping, title, toolbarContainer, colorScaleType } = props;
+  const {
+    value,
+    abscissaParams,
+    ordinateParams,
+    title,
+    toolbarContainer,
+    colorScaleType,
+  } = props;
 
-  const [xAxis, yAxis] = axisMapping;
-
-  assertDefined(xAxis?.value);
-  assertDefined(yAxis?.value);
```

Comment:
Assertions should now done in the HOC as enforced by the change in typings of this component